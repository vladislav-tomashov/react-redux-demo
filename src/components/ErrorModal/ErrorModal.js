import React from "react";
import Modal from "react-modal";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { hideError } from "../../store/actions/errors/errorsActionCreators";
import {
  isErrorShowing,
  getError
} from "../../store/selectors/errorsSelectors";
import "./ErrorModal.scss";

const ErrorModal = ({
  isOpen = false,
  onCloseModal = () => {},
  error = null
} = {}) => (
  <Modal
    isOpen={isOpen}
    className="ErrorModal-modal"
    overlayClassName="ErrorModal-overlay"
    onRequestClose={onCloseModal}
    contentLabel="Error Modal"
  >
    <div className="ErrorModal-title">Error</div>
    {error && <p className="ErrorModal-text">{error.toString()}</p>}
    <button className="ErrorModal-button" onClick={onCloseModal}>
      Ok
    </button>
  </Modal>
);

ErrorModal.propTypes = {
  isOpen: PropTypes.bool,
  error: PropTypes.any,
  onCloseModal: PropTypes.func
};

const mapStateToProps = state => {
  return {
    isOpen: isErrorShowing(state),
    error: getError(state)
  };
};

const mapDispatchToProps = dispatch => {
  return {
    onCloseModal: () => dispatch(hideError())
  };
};

const ConnectedErrorModal = connect(
  mapStateToProps,
  mapDispatchToProps
)(ErrorModal);

export { ConnectedErrorModal as default, ErrorModal };
