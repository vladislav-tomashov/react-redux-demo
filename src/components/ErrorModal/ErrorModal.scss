@import "../../styles/settings";
@import "../PagingButton/PagingButton.scss";

.ErrorModal-overlay {
  align-items: center;
  display: flex;
  justify-content: center;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba($black, 0.6);
}

.ErrorModal-modal {
  background: white;
  border-radius: 4px;
  padding: $l-size $m-size;
  text-align: center;
  min-width: 20rem;
  max-width: 90%;
}

.ErrorModal-text {
  font-size: $font-size-small;
  line-height: 1;
}

.ErrorModal-button {
  @extend .PagingButton;
  font-size: $font-size-small;
  padding: $xs-size;
}

.ErrorModal-title {
  margin: 0 0 $m-size 0;
  line-height: 1;
  font-size: $font-size-xlarge;
  font-weight: 700;
}
