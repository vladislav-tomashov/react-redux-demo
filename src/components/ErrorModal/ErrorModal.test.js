import React from "react";
import { shallow } from "enzyme";
import { ErrorModal } from "./ErrorModal";

let wrapper;

describe("ErrorModal component", () => {
  beforeEach(() => {
    wrapper = shallow(<ErrorModal />);
  });

  test("should render ErrorModal correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should render ErrorModal with all props correctly", () => {
    const hideError = jest.fn();
    wrapper.setProps({
      isErrorShowing: true,
      hideError,
      error: new Error("test error")
    });
    expect(wrapper).toMatchSnapshot();
  });
  test("should handle OK button click", () => {
    const hideError = jest.fn();
    wrapper.setProps({
      isErrorShowing: true,
      hideError,
      error: new Error("test error")
    });
    wrapper.find("button").simulate("click");
    expect(hideError).toBeCalledTimes(1);
  });
});
