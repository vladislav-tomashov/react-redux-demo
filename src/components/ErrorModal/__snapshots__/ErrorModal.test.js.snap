// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorModal component should render ErrorModal correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="ErrorModal-modal"
  closeTimeoutMS={0}
  contentLabel="Error Modal"
  isOpen={false}
  onRequestClose={[Function]}
  overlayClassName="ErrorModal-overlay"
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div
    className="ErrorModal-title"
  >
    Error
  </div>
  <button
    className="ErrorModal-button"
    onClick={[Function]}
  >
    Ok
  </button>
</Modal>
`;

exports[`ErrorModal component should render ErrorModal with all props correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="ErrorModal-modal"
  closeTimeoutMS={0}
  contentLabel="Error Modal"
  isOpen={true}
  onRequestClose={[MockFunction]}
  overlayClassName="ErrorModal-overlay"
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div
    className="ErrorModal-title"
  >
    Error
  </div>
  <p
    className="ErrorModal-text"
  >
    Error: test error
  </p>
  <button
    className="ErrorModal-button"
    onClick={[MockFunction]}
  >
    Ok
  </button>
</Modal>
`;
